spyder/app/tests/test_mainwindow.py

```diff
@@ -862,6 +862,7 @@ def test_connection_to_external_kernel(main_window, qtbot):
     kc.stop_channels()
 
 
+@pytest.mark.first
```

Comment:
Please remove this if it's not really necessary (why your changes affect this test?). We should avoid moving test to be run first whenever possible.

Reply:
>Please remove this if it's not really necessary (why your changes affect this test?).

The changes do not affect the tests directly, but somehow the order in which pytest runs them causes segfaults, that I cannot reproduce locally.

>We should avoid moving test to be run first whenever possible.

This PR will not pass unless that change is made.